packages/vats/test/upgrading/bootstrap.js

```diff
@@ -0,0 +1,64 @@
+import { E, Far } from '@endo/far';
+import { makePromiseKit } from '@endo/promise-kit';
+import { feeIssuerConfig } from '../../src/core/utils';
+
+const bundles = {
+  mintHolder: 'mintHolder',
+};
+
+const terms = { keyword: 'A' /* , assetKind, displayInfo */ };
+
+export const buildRootObject = () => {
+  const zcfBundleName = 'zcf';
+  const bootKit = makePromiseKit();
+  const v1Kit = makePromiseKit();
+
+  return Far('B', {
+    bootstrap: async (vats, devices) => {
+      const vatAdmin = await E(vats.vatAdmin).createVatAdminService(
+        devices.vatAdmin,
+      );
+      const zoeKit = await E(vats.zoe).buildZoe(
+        vatAdmin,
+        feeIssuerConfig,
+        zcfBundleName,
+      );
+
+      const { zoeService: zoe } = zoeKit;
+      const bundleId = await E(vatAdmin).getBundleIDByName(bundles.mintHolder);
+      assert(bundleId, 'bundleId must not be empty');
+      const installation = await E(zoe).installBundleID(bundleId);
+
+      bootKit.resolve({ installation, vatAdmin, zoe, bundleId });
+    },
+
+    startV1: async () => {
+      const { zoe, installation } = await bootKit.promise;
+      const facets = await E(zoe).startInstance(installation, {}, terms);
+
+      const { publicFacet: issuer } = facets;
+      const brand = await E(issuer).getBrand();
+      v1Kit.resolve({ facets, brand });
+      return true;
+    },
+
+    // null upgrade
+    upgradeV1: async () => {
+      const [{ bundleId }, { facets, brand: expectedBrand }] =
+        await Promise.all([bootKit.promise, v1Kit.promise]);
+
+      const upgradeResult = await E(facets.adminFacet).upgradeContract(
+        bundleId,
+      );
+      assert.equal(upgradeResult.incarnationNumber, 2); // move to test assertion?
```

Comment:
makes sense here imo. it's `upgradeV1` checking itself. to use `t` would require letting the test peek in.

Reply:
I thought about returning the `incarnationNumber` and checking in the test code; but returning stuff across the kernel boundary might be tedious.